zephyr/src/utils.c

```diff
@@ -756,6 +756,27 @@ int control_interface(char *ifname, char *op) {
 }
 
 int set_interface_ip(char *ifname, char *ip) {
+
+    const struct device *dev = device_get_binding(ifname);
+    struct net_if *wlan_iface = net_if_lookup_by_dev(dev);
+    struct in_addr addr;
+
+    if (!wlan_iface) {
+	indigo_logger(LOG_LEVEL_ERROR, "Cannot find network interface: %s", dev->name);
+	return -1;
+    }
+
+    if (sizeof(ip) > 1) {
+        if (net_addr_pton(AF_INET, ip, &addr)) {
+            indigo_logger(LOG_LEVEL_ERROR, "Invalid address: %s", ip);
+            return -1;
+        }
+        net_if_ipv4_addr_add(wlan_iface, &addr, NET_ADDR_MANUAL, 0);
```

Comment:
So, while reading we are supporting both IPv4 and IPv6 but while setting only IPv4? Does QT use IPv6 in any of the tests, if not we can just stick to IPv4 to keep things simple.

Reply:
Qt is not using Ipv6 in none of the cases. So will remove reading ipv6 changes.